visitors/es6-class-visitors.js

```diff
@@ -341,15 +341,32 @@ function _renderClassBody(traverse, node, path, state) {
       keyNameDeclarator = 'var ';
       declareIdentInLocalScope(keyName, initScopeMetadata(node), state);
     }
-    utils.append(
-      'for(' + keyNameDeclarator + keyName + ' in ' + superClass.name + '){' +
-        'if(' + superClass.name + '.hasOwnProperty(' + keyName + ')){' +
-          className + '[' + keyName + ']=' +
-            superClass.name + '[' + keyName + '];' +
-        '}' +
-      '}',
-      state
-    );
+    if (state.g.opts.es3) {
+      utils.append(
+        'for(' + keyNameDeclarator + keyName + ' in ' + superClass.name + '){' +
+          'if(' + superClass.name + '.hasOwnProperty(' + keyName + ')){' +
+            className + '[' + keyName + ']=' +
+              superClass.name + '[' + keyName + '];' +
+          '}' +
+        '}',
+        state
+      );
+    } else {
+      var keysName = superClass.name + '____StaticKeys';
+      var keysNameDeclarator = '';
+      if (!utils.identWithinLexicalScope(keysName, state)) {
+        keysNameDeclarator = 'var ';
+        declareIdentInLocalScope(keysName, initScopeMetadata(node), state);
+      }
+      utils.append(
+        keysNameDeclarator + keysName + '=Object.getOwnPropertyNames(' + superClass.name + ').filter(function(key){return ["callee", "caller", "arguments"].indexOf(key)<0;});' +
```

Comment:
This `filter` is gross but `Object.getOwnProperyNames(function() {})` will return ` ["callee", "caller", "arguments])` and any access to those properties on a function in strict mode will throw an exception.